src/shared/organisms/DataExplorerGraphFlowNavigationStack/NavigationStack.spec.tsx

```diff
@@ -0,0 +1,260 @@
+    const state = store.getState();
+    expect(state.dataExplorer.links.length).toBe(4);
+  });
+  it('should render the NavigationStackShrinkedItem when it passed MAX_NAVIGATION_ITEMS_IN_STACK', () => {
+    store.dispatch(
+      AddNewNodeDataExplorerGraphFlow({
+        isDownloadable: false,
+        _self:
+          'https://bbp.epfl.ch/nexus/v1/resources/neurosciencegraph/datamodels/datashapes:ontologyentity/https:%2F%2Fbbp.epfl.ch%2Fontologies%2Fcore%2Fefeatures%2FNeuroElectroNeuronElectrophysiologicalFeature',
+        title: 'NeuroElectro Neuron Electrophysiological Feature',
+        types: ['Class'],
+        resource: [
+          'neurosciencegraph',
+          'datamodels',
+          'https://bbp.epfl.ch/ontologies/core/efeatures/NeuroElectroNeuronElectrophysiologicalFeature',
+          29,
+        ],
+      })
+    );
+    rerender(app);
+    store.dispatch(
+      AddNewNodeDataExplorerGraphFlow({
+        isDownloadable: false,
+        _self:
+          'https://bbp.epfl.ch/nexus/v1/resources/bbp/atlas/datashapes:atlasrelease/4906ab85-694f-469d-962f-c0174e901885',
+        title: 'Blue Brain Atlas',
+        types: ['AtlasRelease', 'BrainAtlasRelease'],
+        resource: [
+          'bbp',
+          'atlas',
+          'https://bbp.epfl.ch/neurosciencegraph/data/4906ab85-694f-469d-962f-c0174e901885',
+          3,
+        ],
+      })
+    );
+    rerender(app);
+    const navigationStackShrinkedItem = container.querySelectorAll(
+      '.navigation-stack-item.more'
+    );
+    expect(navigationStackShrinkedItem.length).toBe(1);
+    expect(navigationStackShrinkedItem).not.toBeNull();
+    const navigationHiddenItems = container.querySelectorAll(
+      '.navigation-stack-item:not(.no-more):not(.more)[hidden]'
+    );
+    expect(navigationHiddenItems.length).toBe(4);
+  });
+  it('should show the collapse button when the NavigationStackShrinkedItem is clicked', () => {
+    const navigationStackShrinkedItem = container.querySelector(
+      '.navigation-stack-item.more'
+    );
+    if (navigationStackShrinkedItem) {
+      user.click(navigationStackShrinkedItem);
+      const collapseBtn = container.querySelector('.navigation-collapse-btn');
+      expect(collapseBtn).not.toBeNull();
+      expect(collapseBtn).toBeInTheDocument();
+      expect(store.getState().dataExplorer.shrinked).toBe(false);
+    }
+  });
+  it('should hide the collapse button when the collapse button is clicked', () => {
+    const navigationCollapseButton = container.querySelector(
+      '.navigation-collapse-btn'
+    );
+    if (navigationCollapseButton) {
+      user.click(navigationCollapseButton);
+      const collapseBtn = container.querySelector('.navigation-collapse-btn');
+      expect(collapseBtn).toBeNull();
+      expect(store.getState().dataExplorer.shrinked).toBe(true);
+    }
+  });
+  it('should the items in the stack be 4 when the user jump to the 5th item', () => {
+    store.dispatch(JumpToNodeDataExplorerGraphFlow(4));
```

Comment:
I'm a bit confused. How is it possible to jump to node 4, if the initialState only has 2 nodes? We should clean the `state` after each test so that the tests don't share any preserved state.

Reply:
If I comment out the test `should render the NavigationStackShrinkedItem when it passed MAX_NAVIGATION_ITEMS_IN_STACK` then the test below for `should the fourth item in the stack will be the current one when the user jump to the 4th item` starts failing. This signals that the tests are right now dependant on the sequence. This can lead to flakiness. We should fix this (by resetting state in beforeEach or cleaning it in afterEach) and make tests more predictable.